prow/git/git_test.go

```diff
@@ -14,17 +14,13 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// this needs to be in git_test to not cause an import cycle
```

Comment:
a) I suspect we can resolve this
b) Put it in a prow/git/testing dir if we cannot and its absolutely necessary

Reply:
I have changed nothing about these tests, just documented why this is the way it is. Do we want a larger diff in this PR? I could tackle this in a separate one if you'd like.